functional/agent/server.go

```diff
@@ -127,10 +127,11 @@ func (srv *Server) Stop() {
 
 // Transport communicates with etcd tester.
 func (srv *Server) Transport(stream rpcpb.Transport_TransportServer) (err error) {
-	errc := make(chan error)
+	errc := make(chan error, 1)
 	go func() {
 		for {
 			var req *rpcpb.Request
+			var err error
```

Comment:
reqerr? and use reqerr in this go-routine? this can make the code more clear.

Reply:
Nope, `err` is used at here `req, err = stream.Recv()` and here `resp, err = srv.handleTesterRequest(req)` but let me change outside `err` L129 to `reterr`.